domains/transfer-agreements/TransferAgreements.API/API/Controllers/TransferAgreementsController.cs

```diff
@@ -37,4 +39,14 @@ public async Task<ActionResult> Create([FromBody] CreateTransferAgreement reques
 
         return Created($"api/transfer-agreements/{result.Id}", result);
     }
+
+    [HttpGet("{id}")]
+    public async Task<ActionResult> Get([FromRoute] Guid id)
+    {
+        var tin = User.FindFirstValue("tin")!;
+        var subject = User.FindSubjectClaim();
+
+        var result = await transferAgreementRepository.GetTransferAgreement(id, subject, tin);
+        return result == null ? NotFound() : Ok(result);
```

Comment:
The OpenAPI/swagger definition states that this endpoint only returns 200 OK without specifying the type returned.
I suggest you add `ProducesResponseType`-attributes, so it states that it will return 
* 200 OK with specific type returned
* 404 Not Found